clang/test/Driver/sycl-specific-args-diagnostics.cpp

```diff
@@ -0,0 +1,140 @@
+// RUN:   %clang -### -fsycl-embed-ir  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-embed-ir %s
+// RUN:   %clang_cl -### -fsycl-embed-ir  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-embed-ir %s
+
+// Warning should be emitted when using -fsycl-esimd-force-stateless-mem without -fsycl
+// RUN:   %clang -### -fsycl-esimd-force-stateless-mem  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-esimd-force-stateless-mem %s
+// RUN:   %clang_cl -### -fsycl-esimd-force-stateless-mem  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-esimd-force-stateless-mem %s
+
+// Warning should be emitted when using -fsycl-force-inline-kernel-lambda without -fsycl
+// RUN:   %clang -### -fsycl-force-inline-kernel-lambda  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-force-inline-kernel-lambda %s
+// RUN:   %clang_cl -### -fsycl-force-inline-kernel-lambda  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-force-inline-kernel-lambda %s
+
+// Warning should be emitted when using -fsycl-fp32-prec-sqrt without -fsycl
+// RUN:   %clang -### -fsycl-fp32-prec-sqrt  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-FP32 -DOPT=-fsycl-fp32-prec-sqrt %s
+// WARNING-FP32: warning: argument unused during compilation: '-fsycl-fp32-prec-sqrt' [-Wunused-command-line-argument]
+// RUN:   %clang_cl -### -fsycl-fp32-prec-sqrt  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-FP32-CL -DOPT=-fsycl-fp32-prec-sqrt %s
+// WARNING-FP32-CL: warning: unknown argument ignored in clang-cl: '[[OPT]]' [-Wunknown-argument]
+
+// Warning should be emitted when using -fsycl-id-queries-fit-in-int without -fsycl
+// RUN:   %clang -### -fsycl-id-queries-fit-in-int  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-id-queries-fit-in-int %s
+// RUN:   %clang_cl -### -fsycl-id-queries-fit-in-int  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-id-queries-fit-in-int %s
+
+// Warning should be emitted when using -fsycl-instrument-device-code without -fsycl
+// RUN:   %clang -### -fsycl-instrument-device-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-instrument-device-code %s
+// RUN:   %clang_cl -### -fsycl-instrument-device-code  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-UNUSED-ARG -DOPT=-fsycl-instrument-device-code %s
+
+// Warning should be emitted when using -fsycl-libspirv-path without -fsycl
+// RUN:   %clang -### -fsycl-libspirv-path=libspirv.bc  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-LIBSPRIV  %s
+// RUN:   %clang_cl -### -fsycl-libspirv-path=libspirv.bc  %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=WARNING-LIBSPRIV %s
+// WARNING-LIBSPRIV: warning: argument unused during compilation: '-fsycl-libspirv-path=libspirv.bc' [-Wunused-command-line-argument]
```

Comment:
Looks like you can use `WARNING-UNUSED-ARG` here

Reply:
These warnings and the others you have mentioned below are emitted with the option and the value, so we are unable to use the generic check-prefix here and at other places.

Reply:
Are you saying because there is an '=' in the string, it cannot be used?  It's just a string check so `OPT` can be whatever you put